tensorflow_probability/python/experimental/mcmc/covariance_reducer_test.py

```diff
@@ -0,0 +1,149 @@
+
+  def test_zero_covariance(self):
+    cov_reducer = tfp.experimental.mcmc.CovarianceReducer()
+    state = cov_reducer.initialize(0.)
+    for _ in range(2):
+      state = cov_reducer.one_step(0., state)
+    final_num_samples, final_mean, final_cov = self.evaluate([
+        state.num_samples,
+        state.mean,
+        cov_reducer.finalize(state)])
+    self.assertEqual(final_num_samples, 2)
+    self.assertEqual(final_mean, 0)
+    self.assertEqual(final_cov, 0)
+
+  def test_random_sanity_check(self):
+    rng = test_util.test_np_rng()
+    x = rng.rand(100)
+    cov_reducer = tfp.experimental.mcmc.CovarianceReducer()
+    state = cov_reducer.initialize(0.)
+    for sample in x:
+      state = cov_reducer.one_step(sample, state)
+    final_mean, final_cov = self.evaluate([
+        state.mean,
+        cov_reducer.finalize(state)])
+    self.assertNear(np.mean(x), final_mean, err=1e-6)
+    self.assertNear(np.var(x, ddof=0), final_cov, err=1e-6)
+
+  def test_covariance_shape(self):
+    cov_reducer = tfp.experimental.mcmc.CovarianceReducer(event_ndims=1)
+    state = cov_reducer.initialize(tf.ones((9, 3)))
+    for _ in range(2):
+      state = cov_reducer.one_step(
+          tf.zeros((5, 9, 3)), state, axis=0)
+    final_mean, final_cov = self.evaluate([
+        state.mean,
+        cov_reducer.finalize(state)])
+    self.assertEqual(final_mean.shape, (9, 3))
+    self.assertEqual(final_cov.shape, (9, 3, 3))
+
+  def test_variance_shape(self):
+    var_reducer = tfp.experimental.mcmc.VarianceReducer()
+    state = var_reducer.initialize(tf.ones((9, 3)))
+    for _ in range(2):
+      state = var_reducer.one_step(
+          tf.zeros((5, 9, 3)), state, axis=0)
+    final_mean, final_var = self.evaluate([
+        state.mean,
+        var_reducer.finalize(state)])
+    self.assertEqual(final_mean.shape, (9, 3))
+    self.assertEqual(final_var.shape, (9, 3))
+
+  def test_attributes(self):
+    cov_reducer = tfp.experimental.mcmc.CovarianceReducer(
+        event_ndims=1, ddof=1)
+    state = cov_reducer.initialize(tf.ones((2, 3), dtype=tf.float64))
+
+    # check attributes are correct right after initialization
+    self.assertEqual(cov_reducer.shape, (2, 3))
+    self.assertEqual(cov_reducer.event_ndims, 1)
+    self.assertEqual(cov_reducer.dtype, tf.float64)
+    self.assertEqual(cov_reducer.ddof, 1)
+    for _ in range(2):
+      state = cov_reducer.one_step(
+          tf.zeros((2, 3)), state)
+
+    # check attributes don't change after stepping through
+    self.assertEqual(cov_reducer.shape, (2, 3))
+    self.assertEqual(cov_reducer.event_ndims, 1)
+    self.assertEqual(cov_reducer.dtype, tf.float64)
+    self.assertEqual(cov_reducer.ddof, 1)
+
+  def test_tf_while(self):
+    cov_reducer = tfp.experimental.mcmc.CovarianceReducer()
+    state = cov_reducer.initialize(tf.ones((2, 3)))
+    _, state = tf.while_loop(
+        lambda i, _: tf.less(i, 100),
+        lambda i, state: (i+1, cov_reducer.one_step(
```

Comment:
Please add spaces around binary ops (here and elsewhere)

Reply:
good catches. Sorry for the repeat style mistake, the code was written before the last comment you made on it and I forgot to fix it outside the scope of that PR.